src/NuGet.Commands/RestoreRequest.cs

```diff
@@ -20,9 +21,35 @@ public RestoreRequest(PackageSpec project, IEnumerable<PackageSource> sources)
         { }
 
         public RestoreRequest(PackageSpec project, IEnumerable<PackageSource> sources, string packagesDirectory)
+            : this(project, packagesDirectory)
         {
+            if (sources == null)
+            {
+                throw new ArgumentNullException(nameof(sources));
+            }
+
+            Sources = sources.Select(source => Repository.Factory.GetCoreV3(source.Source)).ToList();
+        }
+
+        public RestoreRequest(PackageSpec project, IEnumerable<SourceRepository> sources, string packagesDirectory)
```

Comment:
There are too many constructors, with too many ToList calls. Is it possible to simplify it